### ISSUE TITLE
Early stopping with many validation sets

### ISSUE THREAD

**Issue by niknoproblems** [Microsoft/LightGBM#1757]
Have a strange behavior when using early stopping with many validation sets in python package(version 2.2.1)

`clf.fit(X_resampled,y_resampled,early_stopping_rounds=5,eval_set=evals,eval_metric=['auc'])`

**result**:

`[1]	valid_0's auc: 1	valid_1's auc: 1	valid_2's auc: 1	valid_3's auc: 1	valid_4's auc: 1	valid_5's auc: 1	valid_6's auc: 1	valid_7's auc: 1	valid_8's auc: 1	valid_9's auc: 1	valid_10's auc: 1	valid_11's auc: 1	valid_12's auc: 1	valid_13's auc: 0.991215
Training until validation scores don't improve for 5 rounds.

[2]	valid_0's auc: 1	valid_1's auc: 1	valid_2's auc: 1	valid_3's auc: 1	valid_4's auc: 1	valid_5's auc: 1	valid_6's auc: 1	valid_7's auc: 1	valid_8's auc: 1	valid_9's auc: 1	valid_10's auc: 1	valid_11's auc: 1	valid_12's auc: 1	valid_13's auc: 0.993058

[3]	valid_0's auc: 1	valid_1's auc: 1	valid_2's auc: 1	valid_3's auc: 1	valid_4's auc: 1	valid_5's auc: 1	valid_6's auc: 1	valid_7's auc: 1	valid_8's auc: 1	valid_9's auc: 1	valid_10's auc: 1	valid_11's auc: 1	valid_12's auc: 1	valid_13's auc: 0.993104

[4]	valid_0's auc: 1	valid_1's auc: 1	valid_2's auc: 1	valid_3's auc: 1	valid_4's auc: 1	valid_5's auc: 1	valid_6's auc: 1	valid_7's auc: 1	valid_8's auc: 1	valid_9's auc: 1	valid_10's auc: 1	valid_11's auc: 1	valid_12's auc: 1	valid_13's auc: 0.993322

[5]	valid_0's auc: 1	valid_1's auc: 1	valid_2's auc: 1	valid_3's auc: 1	valid_4's auc: 1	valid_5's auc: 1	valid_6's auc: 1	valid_7's auc: 1	valid_8's auc: 1	valid_9's auc: 1	valid_10's auc: 1	valid_11's auc: 1	valid_12's auc: 1	valid_13's auc: 0.993561

[6]	valid_0's auc: 1	valid_1's auc: 1	valid_2's auc: 1	valid_3's auc: 1	valid_4's auc: 1	valid_5's auc: 1	valid_6's auc: 1	valid_7's auc: 1	valid_8's auc: 1	valid_9's auc: 1	valid_10's auc: 1	valid_11's auc: 1	valid_12's auc: 1	valid_13's auc: 0.993599

Early stopping, best iteration is:
[1]	valid_0's auc: 1	valid_1's auc: 1	valid_2's auc: 1	valid_3's auc: 1	valid_4's auc: 1	valid_5's auc: 1	valid_6's auc: 1	valid_7's auc: 1	valid_8's auc: 1	valid_9's auc: 1	valid_10's auc: 1	valid_11's auc: 1	valid_12's auc: 1	valid_13's auc: 0.991215

same with logloss
`

**Comment by guolinke** [comment]
@niknoproblems refer to the document: https://lightgbm.readthedocs.io/en/latest/Parameters.html#early_stopping_round
> will stop training if one metric of one validation data doesn’t improve in last early_stopping_round rounds

since there are many validation sets with auc=1, they trigger the earlystopping. 

@StrikerRUS It seems the document in python package doesn't have these descriptions.

**Comment by niknoproblems** [comment]
Ok, thank you

**Comment by StrikerRUS** [comment]
@guolinke Speaking about sklearn wrapper, we have this notice:
https://github.com/Microsoft/LightGBM/blob/09919669f4d889d76929e13890c9fa78cee9178a/python-package/lightgbm/sklearn.py#L339-L340

`train` and `cv` functions also have it:
https://github.com/Microsoft/LightGBM/blob/09919669f4d889d76929e13890c9fa78cee9178a/python-package/lightgbm/engine.py#L67-L68
https://github.com/Microsoft/LightGBM/blob/09919669f4d889d76929e13890c9fa78cee9178a/python-package/lightgbm/engine.py#L391

The same description can be found in the Python Quick Start:
https://lightgbm.readthedocs.io/en/latest/Python-Intro.html#early-stopping

**Comment by guolinke** [comment]
@StrikerRUS thank you !
